src/v2/Components/BankDebitForm/BankDebitForm.tsx

```diff
@@ -45,20 +52,34 @@ export const BankDebitForm: FC<Props> = ({ order }) => {
 
   const handleSubmit = async event => {
     event.preventDefault()
+    setIsInsufficientFunds(false)
+    setIsConfirmingAccountSetup(true)
+    trackClickedContinue()
 
-    if (!stripe || !elements || !user) {
-      return
-    }
+    if (!stripe || !elements || !user) return
 
-    const { error } = await stripe.confirmSetup({
-      elements,
-      confirmParams: {
-        return_url: "https://staging.artsy.net/",
-      },
-    })
+    try {
+      const { setupIntent, error } = await stripe.confirmSetup({
+        elements,
+        redirect: "if_required",
+        confirmParams: {
+          return_url: "https://staging.artsy.net/",
+        },
+      })
+
+      if (setupIntent?.status === "succeeded") {
+        // TODO: poll account balance and set state accordingly
+        setIsInsufficientFunds(true)
 
-    if (error) {
-      console.log("error", error)
+        router.push(`/orders/${order.internalID}/review`)
```

Comment:
@starsirius what do you think about routing like here, instead of providing stripe with a return_url? (since we want to poll the balance before going to the next step).

Reply:
Sorry for the slowness. We asked Stripe about redirect or not, and they highly recommend us to _not_ use the `if_required` parameter because

- some payment methods require redirect (including some banks for ACH, actually) and if we use `if_required` we will need to build and maintain 2 different user experiences.
- it might disrupt compatibility with future payment methods.

The doc mentions:

> Note: Setting if_required requires that you handle successful confirmations for redirect-based and non-redirect based payment methods separately.